packages/timeline-state-resolver/src/conductor.ts

```diff
@@ -1122,7 +1122,7 @@ export class Conductor extends EventEmitter<ConductorEvents> {
 		)
 	}
 
-	private _diffStateForCallbacks(activeObjects: TimelineCallbacks) {
+	private _diffStateForCallbacks(activeObjects: TimelineCallbacks, tlTime: number) {
 		const sentCallbacks: TimelineCallbacks = this._sentCallbacks
 		const time = this.getCurrentTime()
```

Comment:
Do you still need this since you now parse the time in as a parameter?

Reply:
The way I see it, this is _different_ time. `tlTime` is the time position at which the state has been evaluated. `time` is the current time. That's why callbacks should be cleared, based on a comparison with `time` in L1131, but `_queueCallback` should be at `tlTime` because that's when we want the callback to happen. @nytamin unless you see it differently? I might be misunderstanding the code here.

Reply:
I'm actually leaning towards using `tlTime` on L1131, and not using `getCurrentTime` in this function at all.

My reasoning is this:
`_diffStateForCallbacks()` is queued to be executed at a certain time ([using doOnTime.queue()](https://github.com/nrkno/sofie-timeline-state-resolver/blob/fix/brokenReportedTimings/packages/timeline-state-resolver/src/conductor.ts#L1026)).
Therefore, _if_ `_diffStateForCallbacks()` is executed in time, the two times would be identical. If however, it is not executed in time, we should modify `this._sentCallbacks` using `tlTime`, because that would render the same result, reducing the risk of it being impacted by race-conditions.